### PR TITLE
batch: add functional opts to Batch

### LINKED ISSUE #44: Batch should not hardcodes MaxNodes
The `Batch` type hardcodes MaxNodes to a value of 128 because `The current implementation of flatfs opens this many file descriptors at the same time for the optimized batch write.`

However at this level we should not assume a DAGService is connected to `flatfs` at all.

IMHO `Batch` type should not contain hardcoded defaults and should force the user to provide them on the initialize to make it very explicit that it needs to be a reasonable value for the use of that `Batch` service.

### PR DESCRIPTION
**Pull request by lanzafame**
Resolves #44.

### CODE REVIEW

**Comment by Kubuxu** [comment]
I've started writing about not breaking API compat but this change is source compatible.

File: batch.go [hunk 1]
```diff
@@ -175,3 +173,38 @@ loop:
...
+
+// MaxSizeBatchOption sets the maximum size of a Batch.
+func MaxSizeBatchOption(size int) BatchOption {
+	return func(o *batchOptions) {
+		o.maxSize = size
+	}
+}
+
+// MaxNodesBatchOption sets the maximum number of nodes in a Batch.
+func MaxNodesBatchOption(num int) func(o *batchOptions) {
```
Comment (hsanjuan) [inline batch.go]: shouldn't this return BatchOption ?

Comment (Kubuxu) [reply batch.go]: It should (even thought it would work).

Comment (hsanjuan) [reply batch.go]: Yeah, mostly cosmetic

**Pull request merged**